From bf45f05aa87222b4051c4816dc0325e434a84d7d Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@intevation.de>
Date: Wed, 18 May 2011 19:39:36 +0200
Subject: [PATCH] Disable ServerSide subscriptions by default.

    Enabling Server Side subscriptions in the Accountwizzard
    is a usability problem because it enables an advanced feature
    by default and leaves new users that do no have any
    accounts subscribed with no information why nothing was synced.

    It is now only enabled on mobile devices by default and has
    to be manually enabled on the desktop variants.
---
 resources/imap/imapaccount.cpp             |    4 ++++
 resources/kolabproxy/wizard/kolabwizard.es |    1 -
 2 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/resources/imap/imapaccount.cpp b/resources/imap/imapaccount.cpp
index 0ca4620..f266138 100644
--- a/resources/imap/imapaccount.cpp
+++ b/resources/imap/imapaccount.cpp
@@ -30,7 +30,11 @@ ImapAccount::ImapAccount()
     m_timeout( 30 ),
     m_encryption( KIMAP::LoginJob::Unencrypted ),
     m_authentication( KIMAP::LoginJob::ClearText ),
+#ifdef KDEPIM_MOBILE_UI
+    m_subscriptionEnabled( true )
+#else
     m_subscriptionEnabled( false )
+#endif
 {
 }
 
diff --git a/resources/kolabproxy/wizard/kolabwizard.es b/resources/kolabproxy/wizard/kolabwizard.es
index 3ebd910..1869db4 100644
--- a/resources/kolabproxy/wizard/kolabwizard.es
+++ b/resources/kolabproxy/wizard/kolabwizard.es
@@ -138,7 +138,6 @@ function testOk( arg )
     imapRes.setOption( "AccountIdentity", identity.uoid() );
     imapRes.setOption( "DisconnectedModeEnabled", true );
     imapRes.setOption( "IntervalCheckTime", 60 );
-    imapRes.setOption( "SubscriptionEnabled", true );
     imapRes.setOption( "SieveSupport", true );
     imapRes.setOption( "Authentication", 7 ); // ClearText
     if ( arg == "ssl" ) { 
-- 
1.7.4.4

